pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -514,11 +514,11 @@ func (b *BridgeBindMechanism) startDHCP() error {
 	return nil
 }
 
-func (b *BridgeBindMechanism) preparePodNetworkInterface() error {
+func (b *BridgeBindMechanism) preparePodNetworkInterface() (api.Interface, error) {
```

Comment:
Why would you care at all about the returned interface (nil or not) on case of error ? All errors in this function are non-recoverable, and imply the "death" of the pod.

The only method that receives a pointer is the `setCachedInterface`, and I don't see much sense in attempting to cache a nil value. I'm keeping it as is because it's unrelated to the changes I'm introducing and I'd prefer to keep the get / set cached interface methods symmetric to each other. The load is forced to return a pointer - you don't know if there's a cached iface or not.

I've now split the `preparePodInterface` in two methods; hopefully there's no need to argue this further.